presto-thrift-connector-api/src/main/java/com/facebook/presto/connector/thrift/api/PrestoThriftSplit.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.connector.thrift.api;
+
+import com.facebook.swift.codec.ThriftConstructor;
+import com.facebook.swift.codec.ThriftField;
+import com.facebook.swift.codec.ThriftStruct;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+import static com.facebook.presto.connector.thrift.api.utils.ByteUtils.summarize;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+@ThriftStruct
+public final class PrestoThriftSplit
+{
+    private final byte[] splitId;
```

Comment:
How about we replace `byte[] splitId` and `byte[] nextToken` everywhere with `PrestoThriftId splitId` and ``PrestoThriftId nextToken`.  Then that class internally has the `byte[]` and the `toString` is the code in `ByteUtils` (and we drop the utility package).  This should make usage more safe since it isn't a raw `byte[]` floating around, and the `toString`, `equals`, and `hashCode` would work as expected.